File: src/src_core/fmg/core/img/RotateLogoTransformer.cs
using System;
using fmg.common.geom;
using fmg.common.geom.util;

namespace fmg.core.img {

    /// <summary> Transforming of logo rays </summary>
    public class RotateLogoTransformer : IModelTransformer {

        public void Execute(IAnimatedModel model) {
            if (!(model is LogoModel lm))
                throw new Exception("Illegal usage transformer");

            lm.Rays.Clear();
            lm.Inn.Clear();
            lm.Oct.Clear();

            SizeDouble size = lm.Size;
            var center = new PointDouble(size.Width / 2.0, size.Height / 2.0);
            double ra = lm.RotateAngle;
            FigureHelper.RotateList(lm.Rays, ra, center);
            FigureHelper.RotateList(lm.Inn, ra, center);
            FigureHelper.RotateList(lm.Oct, ra, center);
        }

    }

}

File: src/src_core/fmg/core/img/RotateLogoTransformer.java
package fmg.core.img;

import fmg.common.geom.PointDouble;
import fmg.common.geom.SizeDouble;
import fmg.common.geom.util.FigureHelper;

/** Transforming of logo rays */
public class RotateLogoTransformer implements IModelTransformer {

   @Override
   public void execute(IAnimatedModel model) {
      if (!(model instanceof LogoModel))
         throw new RuntimeException("Illegal usage transformer");

      LogoModel lm = (LogoModel)model;

      lm.getRays().clear();
      lm.getInn().clear();
      lm.getOct().clear();

      SizeDouble size = lm.getSize();
      PointDouble center = new PointDouble(size.width/2.0, size.height/2.0);
      double ra = lm.getRotateAngle();
      FigureHelper.rotateCollection(lm.getRays(), ra, center);
      FigureHelper.rotateCollection(lm.getInn() , ra, center);
      FigureHelper.rotateCollection(lm.getOct() , ra, center);
   }

}
